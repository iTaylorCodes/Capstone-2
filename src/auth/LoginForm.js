import { useState } from "react";
import { useNavigate } from "react-router";
import Alert from "../alert/Alert";
import useFields from "../hooks/useFields";

const LoginForm = ({ login }) => {
  const navigate = useNavigate();
  const [formData, handleChange] = useFields({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  // Handles form submit
  const handleSubmit = async (e) => {
    e.preventDefault();
    let result = await login(formData);
    if (result.success) {
      navigate("/", { replace: true });
    } else {
      setFormErrors(result.errors);
    }
  };

  return (
    <div className="row justify-content-center mb-3">
      <div className="col-10 col-sm-9 col-md-8 col-lg-7 mt-5">
        <h1 className="m-3">Login</h1>
        <div className="card">
          <form onSubmit={handleSubmit}>
            <div className="m-3">
              <label htmlFor="username" className="form-label">
                Username
              </label>
              <input
                id="username"
                name="username"
                value={formData.username}
                onChange={handleChange}
                className="form-control"
                placeholder="Username"
                autoComplete="username"
                required
              />
            </div>
            <div className="m-3">
              <label htmlFor="password" className="form-label">
                Password
              </label>
              <input
                id="password"
                name="password"
                value={formData.password}
                onChange={handleChange}
                type="password"
                className="form-control"
                placeholder="Password"
                autoComplete="current-password"
                required
              />
            </div>

            {formErrors.length ? (
              <Alert type="danger" messages={formErrors} />
            ) : null}

            <div className="m-3 d-flex justify-content-end">
              <button className="btn btn-primary btn-lg" type="submit">
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default LoginForm;
