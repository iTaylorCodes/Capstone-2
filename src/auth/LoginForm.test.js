import { render, fireEvent } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows bootstrap style alerts", function () {
  const { getAllByText } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );

  const button = getAllByText("Login")[1];
  const res = fireEvent.click(button);

  expect(res).toBeTruthy();
});
