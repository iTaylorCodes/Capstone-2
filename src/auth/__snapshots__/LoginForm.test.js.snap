// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center mb-3"
  >
    <div
      class="col-10 col-sm-9 col-md-8 col-lg-7 mt-5"
    >
      <h1
        class="m-3"
      >
        Login
      </h1>
      <div
        class="card"
      >
        <form>
          <div
            class="m-3"
          >
            <label
              class="form-label"
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="username"
              class="form-control"
              id="username"
              name="username"
              placeholder="Username"
              required=""
              value=""
            />
          </div>
          <div
            class="m-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="m-3 d-flex justify-content-end"
          >
            <button
              class="btn btn-primary btn-lg"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
